src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/TalismanListener.java

```diff
@@ -158,14 +158,12 @@ public void onBlockBreak(BlockBreakEvent e) {
 				fortune = (e.getBlock().getType() == Material.LAPIS_ORE ? 4 + random.nextInt(5) : 1) * (fortune + 1);
 			}
 
-			if (!item.getEnchantments().containsKey(Enchantment.SILK_TOUCH) && e.getBlock().getType().toString().endsWith("_ORE") && Talisman.checkFor(e, (SlimefunItemStack) SlimefunItems.TALISMAN_MINER)) {
-				if (drops.isEmpty()) {
-					drops = e.getBlock().getDrops();
-				}
-
-				for (ItemStack drop : new ArrayList<>(drops)) {
+			if (!item.getEnchantments().containsKey(Enchantment.SILK_TOUCH) && MaterialCollections.getAllOres().contains(e.getBlock().getType()) && Talisman.checkFor(e, (SlimefunItemStack) SlimefunItems.TALISMAN_MINER)) {
+				for (ItemStack drop : drops) {
 					if (!drop.getType().isBlock()) {
-						drops.add(new CustomItem(drop, fortune * 2));
+						// Sometimes doubled amount is 0, So make amount at least equals 1
+						int amount = (fortune * 2) - drop.getAmount() == 0 ? 1 : (fortune * 2) - drop.getAmount();
+						e.getPlayer().getInventory().addItem(new CustomItem(drop, amount));
```

Comment:
Adding the items to the inventory directly is a really bad idea.
Just drop them in the world if you really wanna do it that way...
But this will cause problems with full inventories and would be a very inconsistent behaviour.

Reply:
Done.